analyses/cohort_summary/01-generate-cohort-summary-circos-plot.R

```diff
@@ -70,36 +70,48 @@ hist_df <- read_tsv(file.path(data_dir,"histologies.tsv"), guess_max = 100000) %
                                   is.na(cancer_group) & broad_histology == "Tumor of cranial and paraspinal nerves" ~ "Neurofibroma/Plexiform",
                                   TRUE ~ cancer_group))
 
+# find and store all NAs in "age_at_diagnosis_days" that are not PNOC
+hist_NAs_df <- hist_df %>%
+  filter(is.na(age_at_diagnosis_days) & sub_cohort != "PNOC")
+
+# Apply the filter only when sub_cohort is not "PNOC" and remove under 40
+under40_df <- hist_df %>%
+  dplyr::filter(sub_cohort == "PNOC" | 
+           (sub_cohort != "PNOC" & age_at_diagnosis_days < (365.25*40)) |
+              (sub_cohort != "PNOC" & is.na(age_at_diagnosis_days) & age_at_event_days < (365.25*40))
+  )
+
 # add cancer/plot group mapping file 
 map_file <- read_tsv(file.path(input_dir, "plot-mapping.tsv"))
 
 # add plot mapping file and old plot groups, export this.
-combined_plot_map <- hist_df %>%
+combined_plot_map <- under40_df %>%
   full_join(map_file, by = c("broad_histology", "cancer_group")) %>%
   select(names(map_file)) %>%
   unique() %>%
   arrange(broad_histology) %>%
   write_tsv(file.path(results_dir, "plot_mapping.tsv"))
 
 # add plot mapping to histlogy df
-combined_hist_map <- hist_df %>%
+combined_hist_map <- under40_df %>%
   left_join(map_file, by = c("broad_histology", "cancer_group")) %>%
   write_tsv(file.path(results_dir, "histologies-plot-group.tsv"))
 
 combined_hist_map <- combined_hist_map %>%
   filter(experimental_strategy == "RNA-Seq",
          !is.na(pathology_diagnosis))
 
-## filter using independent specimens file
+## filter using independent specimens file 
 independent_specimens_df <- read_tsv(file.path(data_dir,"independent-specimens.rnaseqpanel.primary.tsv")) %>%
   filter(cohort == "PBTA",
-         experimental_strategy == "RNA-Seq")
+         experimental_strategy == "RNA-Seq",
+         Kids_First_Biospecimen_ID %in% combined_hist_map$Kids_First_Biospecimen_ID)
```

Comment:
needed a filter here since you are right joining on indep specimens, else you get the big NA blob in the circos ;)